firebase-common/src/main/java/com/google/firebase/heartbeatinfo/DefaultHeartBeatInfo.java

```diff
@@ -51,10 +60,59 @@ private DefaultHeartBeatInfo(Context context) {
     return HeartBeat.NONE;
   }
 
+  @Override
+  public Task<List<HeartBeatResult>> getAndClearStoredHeartBeatInfo() {
```

Comment:
iiuc you just wrapped the result in a `Task` but did not actually make it asynchronous, so it will block anyway.

Reply:
Oops sorry about that. Updated